### DISCUSSION TITLE
Newest software update has put my laser out of commission

### DISCUSSION THREAD

**Discussion by ShelliTr** [arkypita/LaserGRBL#1210] [Q&A]
So up until the recent update my laser has worked fine on lasergrbl. Now that I have updated it will not work at all.

The motor is running, it is very random when it will actually connect.
When it will connect and I send the file it endlessly file transfers and never actually starts the burn.

I am at a complete loss for what to do, and I just had an order last night come in for some ornaments I make with my laser and I am starting to panic. Please help.

**Comment by hixnhx** [comment]
I have great success with lasergrbl untill the last update and every thing i try to do just burns and doesn't cut.  how do I revert to previous  version

**Comment by arkypita** [comment]
Are you talking about v3.8.5?

**Reply by hixnhx** [reply]
yes v3.8.5

**Comment by arkypita** [comment]
All LaserGRBL version are available here: https://github.com/arkypita/LaserGRBL/releases

**Comment by arkypita** [comment]
The latest v3.8.5 update brings with it an important change: all the code generated by LaserGRBL tools (line2line, dithering, vectorize, centerline, svg) is now homogeneous and uses the same convention already used by line2line and dithering:

This is the convention:
- only one M3/4 command is generated to turn the laser on at first
- all subsequent switching off and on are done by commanding the power (S0 to switch off, S-max to switch on)
- only one M5 command is sent at the end to turn off the laser

This should work correctly with anyone who has a laser that responds correctly to power modulation S commands.
Does your laser support power modulation?

**Comment by arkypita** [comment]
If your laser does not support power modulation simply uncheck the flag "Support PWM" in configuration and LaserGRBL switch back to generate M3/4 - M5 commands to turn on/off the laser (as in the old version).

![image](https://user-images.githubusercontent.com/8782035/102384723-53b2bb80-3fcd-11eb-89c3-f78c0111800e.png)

**Comment by hixnhx** [comment]
I recently installed a ortur 20w laser and it cut card stock perfectly. After upgrading the software and using the same file and same settings the time seemed to double. If 
I just install version 3.80 will things go back to previous version or do I have to uninstall and start over?

**Reply by arkypita** [reply]
The changes to version 3.8.5 were made especially for Ortur (which I own and which I test) and are perfectly compatible with your machine.

You will have noticed in fact that before the fan kept sobbing (it turned on and off during work, in the movements between one cut and another). This update makes the fan run throughout all the job.

The update probably has nothing to do with your problems, nor with time doubled, but if you have doubts just install version 3.8.0 and remove the doubt.

**Reply by arkypita** [reply]
Ps. Ortur is full PWM support so DO NOT UNCHECK "Support PWM" configuration.

**Comment by hixnhx** [comment]
I just bought the laser unit not the machine using a Spark-Concept xpro controller if that has anything to do with my problem. 
Thanks

**Reply by arkypita** [reply]
Spark-Concept xpro controller look like to be PWM compatible, but I have never tried it and therefore I cannot tell you more. It also depends on how you connected the laser to it.
 
Before making other hypotheses, try installing the previous version of LaserGRBL and tell me if something changes.
https://github.com/arkypita/LaserGRBL/releases/download/v3.8.0/install.exe

**Comment by ShelliTr** [comment]
My model is a Uttiny CNC Laser

It is very basic
I have never had these problems
I have replugged all the wires and everything just to be sure it wasnt a problem.
I have to resinstall drivers every single time just to get it to connect now, and whenever I put my image in to start burning it, it just buffers endlessly. Like when I click the start button it never starts burning. Just buffers.

![Screenshot (2)](https://user-images.githubusercontent.com/76114472/102388460-c442e380-3f9f-11eb-889a-fec5b5d9501b.png)

**Comment by ShelliTr** [comment]
![Screenshot (3)](https://user-images.githubusercontent.com/76114472/102388754-2f8cb580-3fa0-11eb-911c-7217e1b094a9.png)

**Reply by arkypita** [reply]
Hi @ShelliTr 
Your problem also doesn't seem related to the update, but if you have any doubt you can download the old version from here: https://github.com/arkypita/LaserGRBL/releases/download/v3.8.0/install.exe

From the picture you sent it seems that the machine does not respond to commands. (the green tick replaces the watch, in command list, when the machine reply with an "ok" message to LaserGRBL command).

> I have to resinstall drivers every single time

This is also a very strange thing, and I would bet my underpants that it has nothing to do with the LaserGRBL update (I have not changed anything regarding these things). 

Why do you have to reinstall the drivers? The COM port no longer appears when you connect it?

**Reply by ShelliTr** [reply]
Well the COMports are there but its like the machine wont recognize that it needs to connect when i put it on comport 6

**Reply by arkypita** [reply]
If you go to grbl->grbl configuration when connected and idle, are you able to read the grbl config from the machine?
(hit refresh and see success read message)

![image](https://user-images.githubusercontent.com/8782035/102390734-17835900-3fd5-11eb-90ec-b808c069f895.png)

**Reply by ShelliTr** [reply]
So whenever I try to reconfigure my settings on that page and I put in the numbers to match yours, and hit write, it starts thinking and then it stops responding.

**Reply by arkypita** [reply]
> So whenever I try to reconfigure my settings on that page and I put in the numbers to match yours, and hit write, it starts thinking and then it stops responding. 

But I didn't tell you to copy and write values like mine. I asked you if you could read yours.
These data are stored in the machine, and are specific and different for each hardware, mine doesn't necessarily work for yours too.

**Reply by arkypita** [reply]
are yours very different from mine? can you post a screenshoot?

**Reply by arkypita** [reply]
any news?

**Comment by hixnhx** [comment]
I reinstalled 3.80 and things seem to be fine. Just did the bottom of a bowl I turned, attached is a sample. Thanks
![20201218_153251](https://user-images.githubusercontent.com/42501875/102664486-81703f80-4148-11eb-88db-510ae0e2d26b.jpg)
